armory/datasets/README.md

```diff
@@ -0,0 +1,76 @@
+# Setup
+
+This is designed to be run in an armory pytorch container.
+Recommended use:
+```
+armory launch pytorch
+```
+Then work in that environment.
+
+# Structure
+
+The code exists in `armory/datasets`, with two subdirectories:
+- `standard` - non-adversarial datasets
+- `adversarial` - adversarial datasets
+
+## Create a new dataset for integration into armory
+
+See the [TFDS v4 CLI](https://www.tensorflow.org/datasets/cli).
+
+On the command line, to create a new standard dataset with name `"my_dataset"`, do:
+```
+cd armory/datasets/standard
+tfds new my_dataset
+```
+This will populate the `armory/datasets/standard/my_dataset` directory.
+
+At a minimum, you will need to fill in the `my_dataset.py` and `checksums.tsv` files.
+
+# Examples
+
+## TFDS standard dataset
+
+MNIST is a standard built-in TFDS dataset. To build this, do:
+```
+python -m armory.datasets.build mnist
+```
+
+To load it, you can do:
+```
+python -m armory.datasets.load mnist
+```
+
+Or, in an interpreter, you can do:
+```
+from armory.datasets import load
+info, ds = load.load("mnist")
+# Alternatively:
```

Comment:
Overall this looks pretty neat, and I want to take a closer look at what the code is doing. One issue I did hit while following the tutorial was attempting
```
>>> info, ds = load.from_directory("/armory/datasets/new_builds/mnist/3.0.1")
```

I ran into the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/workspace/armory/datasets/load.py", line 9, in from_directory
    builder = tfds.builder_from_directory(builder_dir)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/read_only_builder.py", line 130, in builder_from_directory
    return ReadOnlyBuilder(builder_dir=builder_dir)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/read_only_builder.py", line 84, in __init__
    raise ValueError(
ValueError: Cannot restore mnist/3.0.1. It likely means the dataset was generated with an old TFDS version (<=3.2.1).
```

which was odd since I did initially confirm once in the container that I was using tfdsv4:
```
I have no name!@c6c4b7a87ae8:/workspace$ tfds --version
2022-11-02 17:37:28.496520: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-11-02 17:37:28.631918: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2022-11-02 17:37:29.254790: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 17:37:29.254857: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 17:37:29.254868: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
TensorFlow Datasets: 4.6.0

```

To build the dataset, I had used 
```
python -m armory.datasets.build mnist
```

Are you able to use `from_directory()` successfully?

Reply:
Not with mnist. It looks like for whatever reason the mnist builder doesn't populate a `features.json` file, which is needed by TFDS v4. Did that function work with digit for you?

Reply:
I get the following error when trying to build digit, is this the same error you were seeing?

```
I have no name!@c6c4b7a87ae8:/workspace$ python -m armory.datasets.build digit
2022-11-02 18:50:39.072928: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-11-02 18:50:39.212006: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2022-11-02 18:50:39.813876: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 18:50:39.813931: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 18:50:39.813940: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
Executing: tfds build /workspace/armory/datasets/standard/digit --data_dir /armory/datasets/new_builds --force_checksums_validation
2022-11-02 18:50:41.227187: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-11-02 18:50:41.358843: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2022-11-02 18:50:41.963006: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 18:50:41.963061: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 18:50:41.963071: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2022-11-02 18:50:43.254609: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/
2022-11-02 18:50:43.254639: W tensorflow/stream_executor/cuda/cuda_driver.cc:263] failed call to cuInit: UNKNOWN ERROR (303)
2022-11-02 18:50:43.254656: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:163] no NVIDIA GPU device is present: /dev/nvidia0 does not exist
INFO[build.py]: Loading dataset /workspace/armory/datasets/standard/digit from path: /workspace/armory/datasets/standard/digit/digit.py
2022-11-02 18:50:43.429595: W tensorflow/core/platform/cloud/google_auth_provider.cc:184] All attempts to get a Google authentication bearer token failed, returning an empty token. Retrieving token from files failed with "NOT_FOUND: Could not locate the credentials file.". Retrieving token from GCE failed with "FAILED_PRECONDITION: Error executing an HTTP request: libcurl code 6 meaning 'Couldn't resolve host name', error details: Could not resolve host: metadata".
INFO[build.py]: download_and_prepare for dataset digit/1.0.8...
INFO[dataset_builder.py]: Generating dataset digit (/armory/datasets/new_builds/digit/1.0.8)
Downloading and preparing dataset Unknown size (download: Unknown size, generated: Unknown size, total: Unknown size) to /armory/datasets/new_builds/digit/1.0.8...
INFO[download_manager.py]: Downloading https://todo-data-url into /armory/datasets/new_builds/downloads/todo-data-urlpBUd8qQ23lt5lhA0m7PEVFkuiX1TnaXfmqZ3TdfbiFU.tmp.8da80d48a11d40918f2d26ccc49a3e0b...
Extraction completed...: 0 file [00:00, ? file/s]                                                                           | 0/1 [00:00<?, ? url/s]
Dl Size...: 0 MiB [00:00, ? MiB/s]
Dl Completed...:   0%|                                                                                                      | 0/1 [00:00<?, ? url/s]
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/opt/conda/lib/python3.9/site-packages/urllib3/util/connection.py", line 72, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/opt/conda/lib/python3.9/socket.py", line 954, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connectionpool.py", line 386, in _make_request
    self._validate_conn(conn)
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1040, in _validate_conn
    conn.connect()
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connection.py", line 358, in connect
    conn = self._new_conn()
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f27e700f5e0>: Failed to establish a new connection: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/opt/conda/lib/python3.9/site-packages/urllib3/connectionpool.py", line 785, in urlopen
    retries = retries.increment(
  File "/opt/conda/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='todo-data-url', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f27e700f5e0>: Failed to establish a new connection: [Errno -2] Name or service not known'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/tfds", line 8, in <module>
    sys.exit(launch_cli())
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/main.py", line 102, in launch_cli
    app.run(main, flags_parser=_parse_flags)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/main.py", line 97, in main
    args.subparser_fn(args)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/build.py", line 192, in _build_datasets
    _download_and_prepare(args, builder)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/build.py", line 342, in _download_and_prepare
    builder.download_and_prepare(
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/dataset_builder.py", line 481, in download_and_prepare
    self._download_and_prepare(
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/dataset_builder.py", line 1184, in _download_and_prepare
    split_generators = self._split_generators(  # pylint: disable=unexpected-keyword-arg
  File "/workspace/armory/datasets/standard/digit/digit.py", line 74, in _split_generators
    path = dl_manager.download_and_extract("https://todo-data-url")
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/download/download_manager.py", line 633, in download_and_extract
    return _map_promise(self._download_extract, url_or_urls)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/download/download_manager.py", line 766, in _map_promise
    res = tf.nest.map_structure(lambda p: p.get(), all_promises)  # Wait promises
  File "/opt/conda/lib/python3.9/site-packages/tensorflow/python/util/nest.py", line 917, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/opt/conda/lib/python3.9/site-packages/tensorflow/python/util/nest.py", line 917, in <listcomp>
    structure[0], [func(*x) for x in entries],
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/download/download_manager.py", line 766, in <lambda>
    res = tf.nest.map_structure(lambda p: p.get(), all_promises)  # Wait promises
  File "/opt/conda/lib/python3.9/site-packages/promise/promise.py", line 512, in get
    return self._target_settled_value(_raise=True)
  File "/opt/conda/lib/python3.9/site-packages/promise/promise.py", line 516, in _target_settled_value
    return self._target()._settled_value(_raise)
  File "/opt/conda/lib/python3.9/site-packages/promise/promise.py", line 226, in _settled_value
    reraise(type(raise_val), raise_val, self._traceback)
  File "/opt/conda/lib/python3.9/site-packages/six.py", line 719, in reraise
    raise value
  File "/opt/conda/lib/python3.9/site-packages/promise/promise.py", line 844, in handle_future_result
    resolve(future.result())
  File "/opt/conda/lib/python3.9/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/opt/conda/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/opt/conda/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/download/downloader.py", line 217, in _sync_download
    with _open_url(url, verify=verify) as (response, iter_content):
  File "/opt/conda/lib/python3.9/contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/download/downloader.py", line 278, in _open_with_requests
    with session.get(url, stream=True, **kwargs) as response:
  File "/opt/conda/lib/python3.9/site-packages/requests/sessions.py", line 542, in get
    return self.request('GET', url, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/conda/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/requests/adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='todo-data-url', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f27e700f5e0>: Failed to establish a new connection: [Errno -2] Name or service not known'))
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/conda/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/workspace/armory/datasets/build.py", line 143, in <module>
    build(
  File "/workspace/armory/datasets/build.py", line 108, in build
    return build_armory_dataset(
  File "/workspace/armory/datasets/build.py", line 69, in build_armory_dataset
    subprocess.run(cmd, check=True)
  File "/opt/conda/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['tfds', 'build', '/workspace/armory/datasets/standard/digit', '--data_dir', '/armory/datasets/new_builds', '--force_checksums_validation']' returned non-zero exit status 1.

```

Reply:
Yes, that's it. I don't know why it's a problem - something with s3?

Reply:
I wonder if this is related:
```
I have no name!@335fff5e9376:/workspace$ tfds build digit                                                                               
2022-11-02 19:35:13.913377: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Ne
ural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F AVX512_VNNI FMA   
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.                                             
2022-11-02 19:35:14.014784: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerica
l results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_E
NABLE_ONEDNN_OPTS=0`.                                                                                                                   
2022-11-02 19:35:14.041407: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to regist
er factory for plugin cuBLAS when one has already been registered                                                                       
2022-11-02 19:35:14.700566: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.s
o.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/l
ocal/nvidia/lib64:/opt/conda/lib/                                                                                                       
2022-11-02 19:35:14.700618: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_p
lugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvi
dia/lib:/usr/local/nvidia/lib64:/opt/conda/lib/                                                                                         
2022-11-02 19:35:14.700625: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT librarie
s. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.    
2022-11-02 19:35:15.909121: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1
'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/n
vidia/lib64:/opt/conda/lib/                                                                                                             
2022-11-02 19:35:15.909156: W tensorflow/stream_executor/cuda/cuda_driver.cc:263] failed call to cuInit: UNKNOWN ERROR (303)            
2022-11-02 19:35:15.909180: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:163] no NVIDIA GPU device is present: /dev/nvidia0 doe
s not exist                                                                                                                             
Traceback (most recent call last):                                                                                                      
  File "/opt/conda/bin/tfds", line 8, in <module>                                                                                       
    sys.exit(launch_cli())                                                                                                              
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/main.py", line 102, in launch_cli                        
    app.run(main, flags_parser=_parse_flags)                                                                                            
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 308, in run                                                           
    _run_main(main, args)                                                                                                               
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 254, in _run_main                                                     
    sys.exit(main(argv))                                                                                                                
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/main.py", line 97, in main                               
    args.subparser_fn(args)                                                                                                             
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/build.py", line 191, in _build_datasets                  
    for builder in _make_builders(args, ds_to_build):                                                                                   
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/build.py", line 200, in _make_builders                   
    builder_cls, builder_kwargs = _get_builder_cls(ds_to_build)                                                                         
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/scripts/cli/build.py", line 261, in _get_builder_cls                 
    builder_cls = tfds.builder_cls(str(name))                                                                                           
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/load.py", line 101, in builder_cls
    _reraise_with_list_builders(e, name=ds_name)  # pytype: disable=bad-return-type
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/load.py", line 454, in _reraise_with_list_builders
    raise py_utils.reraise(e, suffix=error_string)
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/load.py", line 97, in builder_cls
    cls = registered.imported_builder_cls(str(ds_name))
  File "/opt/conda/lib/python3.9/site-packages/tensorflow_datasets/core/registered.py", line 129, in imported_builder_cls
    raise DatasetNotFoundError(f'Dataset {name} not found.')
tensorflow_datasets.core.registered.DatasetNotFoundError: Dataset digit not found.
Available datasets:

[insert long list that doesn't include digit]

Check that:
    - if dataset was added recently, it may only be available
      in `tfds-nightly`
    - the dataset name is spelled correctly
    - dataset class defines all base class abstract methods
    - the module defining the dataset class is imported

```

Granted, it's clear that `"digit" in common.tfds_builders()` is `True` [here](https://github.com/davidslater/armory/blob/build-datasets/armory/datasets/build.py#L114), but it feels like it may not be a coincidence that I'm seeing this error using tfds cli

Reply:
Actually perhaps not:
```
>>> "mnist" in tfds.list_builders()
True
>>> "digit" in tfds.list_builders()
False

```

Reply:
This is what `build` is calling on the command line for `digit`:
```
tfds build /workspace/armory/datasets/standard/digit --data_dir /armory/datasets/new_builds --force_checksums_validation
```

Reply:
The issue is with [this](https://github.com/davidslater/armory/blob/build-datasets/armory/datasets/standard/digit/digit.py#L74) line. Once you change that to point to `_URL`, that error goes away

At this point, I then get a checksum error:
```
ValueError: Missing checksums url: https://armory-public-data.s3.us-east-2.amazonaws.com/digit/digit-1.0.8.tar.gz, yet `force_checksums_validation=True`. Did you forget to register checksums?

```

when can then be fixed by adding the `--register_checksum` flag as such:

```
python -m armory.datasets.build digit --register_checksums
```

^the above call (after fixing the url) builds the dataset successfully

Reply:
Haha, nice catch! Definitely removed the wrong line.

Reply:
Pushed the fix, updated README, and updated checksums for digit